### DISCUSSION TITLE
Deleting a ContentType and instances programmatically

### DISCUSSION THREAD

**Discussion by kevinsmgov** [OrchardCMS/OrchardCore#10279] [Q&A] [answered]
In a migrations file, I have the option of deleting an existing content type . However, just deleting the type does not delete instances of the type, leaving their routes reserved.

What would be the recommended way to accomplish this in a migrations file (i.e. so when the site is deployed to a production server, the content type and any instances are all removed automatically)?

**Comment by Skrypt** [comment, marked as answer]
I would simply use a SQL script for this.

**Reply by kevinsmgov** [reply]
In the migrations file?

My use case is that I'm replacing a page with embedded scripting with a controller. When I deploy to production, the cutover needs to happen immediately (i.e. visitors hitting the previous page's route need to go to the controller instead)

**Reply by kevinsmgov** [reply]
That worked! Thanks

**Reply by Skrypt** [reply]
Yes, the controller should work as soon as your module is enabled in that case. So updating the SQL data afterward might just work. Though, I strongly advise to restart the application pool once the migration is done to refresh the cache entries. So, it might be better to do the SQL migration before updating the website depending on how you are hosting your website. Docker is always better for doing these kind of things.
